Lombiq.Privacy.Tests.UI/Extensions/TestCaseUITestContextExtensions.cs

```diff
@@ -0,0 +1,120 @@
+using Lombiq.Privacy.Tests.UI.Constants;
+using Lombiq.Tests.UI.Extensions;
+using Lombiq.Tests.UI.Services;
+using OpenQA.Selenium;
+using OrchardCore.Users.Models;
+using Shouldly;
+using System.Threading.Tasks;
+
+namespace Lombiq.Privacy.Tests.UI.Extensions;
+
+public static class TestCaseUITestContextExtensions
+{
+    public static async Task TestPrivacySampleBehaviorAsync(this UITestContext context)
+    {
+        await context.ExecutePrivacySampleRecipeDirectlyAsync();
+
+        await context.TestFormConsentCheckboxContentAsync();
+
+        await context.GoToRelativeUrlAsync(AbsolutePaths.CompetitorRegistration);
+
+        // Here we fill up the competitor form.
+        await context.FillInWithRetriesAsync(By.Id("full-name"), TestCompetitor.FullName);
+        await context.FillInWithRetriesAsync(By.Id("age"), TestCompetitor.Age);
+        await context.FillInWithRetriesAsync(By.Id("e-mail"), TestCompetitor.Email);
+        await context.ClickReliablyOnSubmitAsync();
+
+        // Consent checkbox left unchecked, so after clicking on submit navigation should not happens.
+        var currentUri = context.GetCurrentUri();
+        currentUri.AbsolutePath
+            .ShouldBe(AbsolutePaths.CompetitorRegistration);
```

Comment:
You don't really need a variable here.
```suggestion
        context.GetCurrentUri().ShouldBe(AbsolutePaths.CompetitorRegistration);
```